packages/core/src/services/invitations/validateInvitation.ts

```diff
@@ -0,0 +1,34 @@
+import { and, eq, gte } from 'drizzle-orm'
+
+import { database } from '../../client'
+import { Result, TypedResult } from '../../lib/Result'
+import { Invitation, invitations } from '../../schema/models/invitations'
+
+interface ValidateInvitationArgs {
+  token: string
+}
+
+export async function validateInvitation(
+  { token }: ValidateInvitationArgs,
+  db = database,
+): Promise<TypedResult<Invitation, Error>> {
+  try {
+    const invitation = await db.query.invitations.findFirst({
+      where: and(
+        eq(invitations.token, token),
+        eq(invitations.status, 'pending'),
```

Comment:
Instead of a `status` do a timestamp `aceptedAt` that is `null` or `Date`. This way is tracked when user accepted the inviation.